Ameba-D SDK: lwIP arch/cc.h 의 timeval / in_addr_t 를 표준 newlib 기준으로

문제
----
port/realtek/arch/cc.h 가 CONFIG_PLATFORM_8721D + GCC>=6 이면
LWIP_TIMEVAL_PRIVATE 0 과 IN_ADDR_T_DEFINED 1 을 켠다. 벤더 asdk newlib 이
struct timeval 과 in_addr_t 를 제공한다는 전제다.

이 저장소는 표준 arm-none-eabi-gcc 로 빌드한다. 그 newlib 에는 in_addr_t 가
없고 struct timeval 도 자동으로 따라오지 않아 이렇게 깨진다:

    lwip/inet.h:59:3: error: unknown type name 'in_addr_t'
    lwip/sockets.h:507:24: warning: 'struct timeval' declared inside parameter list
    src/api/sockets.c:1371:1: error: conflicting types for 'lwip_select'

해결
----
조건에 CONFIG_ASDK_NEWLIB 를 추가한다. 정의하지 않으면 lwIP 가 자기
struct timeval 과 in_addr_t 를 쓴다. asdk 툴체인으로 빌드할 때만 정의하면
원래 동작으로 돌아간다.

--- a/network/lwip/port/realtek/arch/cc.h
+++ b/network/lwip/port/realtek/arch/cc.h
@@ -100,7 +100,13 @@
 
 #define LWIP_NO_STDINT_H 1
 
-#if (defined (CONFIG_PLATFORM_8710C)||defined(CONFIG_PLATFORM_8721D)) && defined (__GNUC__) && (__GNUC__ >= 6)
+/* 벤더 asdk newlib 은 struct timeval 과 in_addr_t 를 제공하지만 표준
+ * arm-none-eabi newlib 에는 in_addr_t 가 없다. 이 저장소는 표준 툴체인으로
+ * 빌드하므로 lwIP 가 자기 정의를 쓰게 둔다.
+ * asdk 로 빌드할 때만 CONFIG_ASDK_NEWLIB 를 정의한다. */
+#if defined(CONFIG_ASDK_NEWLIB) && \
+    (defined (CONFIG_PLATFORM_8710C)||defined(CONFIG_PLATFORM_8721D)) && \
+    defined (__GNUC__) && (__GNUC__ >= 6)
 #define LWIP_TIMEVAL_PRIVATE 0
 #define IN_ADDR_T_DEFINED 1
 #endif
